Cordial Actions destination doc (#2870)
segmentio/segment-docs: src/connections/destinations/catalog/actions-cordial/index.md

- **Sending Personas components to Cordial**. With Cordial (Actions) it is possible to define action mappings that will send audiences and user computed traits defined in Segment Personas platform to Cordial.

## Getting started

1. To enable destination actions to connect to Cordial, you will need a Cordial API Key. API keys can be created on the API keys page page the Account Settings menu. [Learn more](https://support.cordial.com/hc/en-us/articles/115005365087)
2. From the Segment web app, click **Catalog**, then click **Destinations**.
3. Find the Destinations Actions item in the left navigation, and click it.
4. Click **Configure Cordial**.
5. Select an existing Source to connect to Cordial (Actions).

<!-- The line below renders a table of connection settings (if applicable), Pre-built Mappings, and available actions. -->

{% include components/actions-fields.html %}

## Migration from the classic Cordial destination

### User Identities Mappings

> Every Cordial destination action needs to be invoked with data identifying a Cordial contact. To identify a contact every destination action has a `User Identities` field, which maps Segment event fields to Cordial contact identifiers. Each entry in the list represents a contact identifier and how it maps from a Segment event. For example, `channels.email.address <- userId` or `customerId <- traits.customerId`. At least one identifier should be valid, otherwise the contact will not be identified and the request will be ignored.

Reviewer: ```suggestion
Every Cordial destination action needs to be invoked with data identifying a Cordial contact. To identify a contact every destination action has a `User Identities` field, which maps Segment event fields to Cordial contact identifiers. Each entry in the list represents a contact identifier and how it maps from a Segment event. For example, `channels.email.address <- userId` or `customerId <- traits.customerId`. At least one identifier should be valid, otherwise the contact won't be identified and the request will be ignored.
```

> Typically `User Identities` field will map Segment event's `userId` field to Cordial secondary identifier field. For example, if Segment's `userId` field is known to be an email, the mapping will be `channels.email.address <- userId`, meaning the value of `userId` will be sent as `channels.email.address` to Cordial.

Reviewer: ```suggestion
Typically, the `User Identities` field maps the Segment events `userId` field to the Cordial secondary identifier field. For example, if Segment's `userId` field is known to be an email, the mapping will be `channels.email.address <- userId`, meaning the value of `userId` will be sent as `channels.email.address` to Cordial.
```

### Updating contacts

> If you plan to create and update contacts in Cordial, define `upsertContact` destination action. In addition to `User Identities` field, the action defines `Contact Attributes` field. This field defines an exclusive set of attributes that will be updated in a contact. Typically you map them from Segment traits, for example, `customerId <- traits.customerId`. For Cordial Classic destination these mappings are stored in Cordial's database, for Cordial (Actions) destination they become explicit in the `upsertContact` destination action mappings.

Reviewer: ```suggestion
If you plan to create and update contacts in Cordial, define the `upsertContact` destination action. In addition to the `User Identities` field, the action defines the `Contact Attributes` field. This field defines an exclusive set of attributes that will be updated in a contact. Typically, you map them from Segment traits. For example, `customerId <- traits.customerId`. For the Cordial Classic destination, these mappings are stored in Cordial's database. In the Cordial (Actions) destination, they become explicit in the `upsertContact` destination action mappings.
```

### Sending events

> To send events define `createContactactivity` destination action. In addition to `User Identities` field, additional fields such as `Event name`, `Event timestamp`, and `Event properties` may be defined. Refer to documentation for each field when defining the destination action.

Reviewer: ```suggestion
To send events, define the `createContactactivity` destination action. In addition to the `User Identities` field, additional fields such as `Event name`, `Event timestamp`, and `Event properties` may be defined. Refer to documentation for each field when defining the destination action.
```

### Adding users to and removing from lists

> If you plan segmenting users in Cordial, make sure you define `addContactToList` and `removeContactFromList` destination actions. Both actions require Segment group ID. `addContactToList` optionally accepts a list name. 

Reviewer: ```suggestion
If you plan segmenting users in Cordial, make sure you define the `addContactToList` and `removeContactFromList` destination actions. Both actions require the Segment group ID. `addContactToList` optionally accepts a list name. 
```

> Although optional, list name should be considered a must have, because it will drastically simplify segmenting contacts in Cordial based on lists. Lists without a name are called following the `segment_[groupId]` pattern.

Reviewer: ```suggestion
Although optional, list name should be considered a must have, because it simplifies segmenting contacts in Cordial based on lists. Lists without a name are called following the `segment_[groupId]` pattern.
```
